v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/dialectadapter/mysql/MysqlDialectAdapter.java

```diff
@@ -376,6 +376,7 @@ private ImmutableMap<String, SourceColumnType> getTableCols(
           .put("TINYTEXT", IndexType.STRING)
           .put("DATETIME", IndexType.TIME_STAMP)
           .put("TIMESTAMP", IndexType.TIME_STAMP)
+          .put("DATE", IndexType.TIME_STAMP)
```

Comment:
There's a difference of precision between Date and Timestamp.
`ResultSet::getTimestamp` yields a timestamp in milli seconds precision. 
`ResultSet::getDate` yields a date which is precise only to a day.

When we map a date to the more precise timestamp and split it, the resultant timestamp is also more precise.

For example a range of 2025/12/10 and 2025/12/11 is unsplitable but when mapped to a timestamp they it would unnecessarily split further.

For `Date` it might make sense to add an indexType called Date and extract and split boundary accordingly.

For BoundayExtraction you can use `getDate` and for splitting you can map the extracted `Date`  to number of days since Epoch and then split the mapped value and unmap it back to `Date`